File: solutions/1641/1641-kir3i.py
class Solution:
    f = {}
    def countVowelStrings(self, n):
        return self.nHr(5, n)

    def nHr(self, n, r):
        return self.fact(n+r-1) // self.fact(n-1) // self.fact(r)

    def fact(self, n):
        if n <= 1:
            return 1
        elif n not in self.f:
            self.f[n] = n * self.fact(n-1)
        return self.f[n]

File: solutions/1641/1641-kir3i.cpp
class Solution {
public:
    int countVowelStrings(int n) {
        return nHr(5, n);
    }
    
    int nHr(int n, int r) {
        double ans = 1.;
        for(int i=0; i<r; i++)
            ans *= 1.0 * (n+r-1-i) / (i+1);
        return round(ans);
    }
};